engine/access/rpc/backend/script_executor_test.go

```diff
@@ -0,0 +1,293 @@
+	})
+
+	s.Run("test script execution with version control with compatible version", func() {
+		// Set up a mock version beacons storage.
+		versionBeacons := storageMock.NewVersionBeacons(s.T())
+		versionEvents := map[uint64]*flow.SealedVersionBeacon{
+			s.height: versionBeaconEvent(
+				s.T(),
+				s.height,
+				[]uint64{s.height},
+				[]string{"0.0.1"},
+			),
+		}
+		// Mock the Highest method to return a version beacon with a specific version.
+		versionBeacons.
+			On("Highest", testifyMock.AnythingOfType("uint64")).
+			Return(mocks.StorageMapGetter(versionEvents))
+
+		var err error
+		s.versionControl, err = version.NewVersionControl(
+			s.log,
+			versionBeacons,
+			semver.New("0.0.1"),
+			s.height-1,
+			s.height,
+		)
+		require.NoError(s.T(), err)
+
+		// Create a mock signaler context for testing.
+		ictx := irrecoverable.NewMockSignalerContext(s.T(), ctx)
+
+		// Start the VersionControl component.
+		s.versionControl.Start(ictx)
+
+		// Ensure the component is ready before proceeding.
+		unittest.RequireComponentsReadyBefore(s.T(), 2*time.Second, s.versionControl)
+
+		scriptExec := NewScriptExecutor(s.log, uint64(0), math.MaxUint64)
+		s.reporter.On("HighestIndexedHeight").Return(s.height+1, nil)
+
+		err = scriptExec.Initialize(s.indexReporter, s.scripts, s.versionControl)
+		s.Require().NoError(err)
+
+		res, err := scriptExec.ExecuteAtBlockHeight(ctx, script, scriptArgs, s.height)
+		s.Assert().NoError(err)
+		s.Assert().NotNil(res)
+		s.Assert().Equal(expectedResult, res)
+	})
+
+	s.Run("test script execution with version control with incompatible version", func() {
+		// Set up a mock version beacons storage.
+		versionBeacons := storageMock.NewVersionBeacons(s.T())
+		versionEvents := map[uint64]*flow.SealedVersionBeacon{
+			s.height: versionBeaconEvent(
+				s.T(),
+				s.height,
+				[]uint64{s.height},
+				[]string{"0.0.2"},
+			),
+		}
+		// Mock the Highest method to return a version beacon with a specific version.
+		versionBeacons.
+			On("Highest", testifyMock.AnythingOfType("uint64")).
+			Return(mocks.StorageMapGetter(versionEvents))
+
+		var err error
+		s.versionControl, err = version.NewVersionControl(
+			s.log,
+			versionBeacons,
+			semver.New("0.0.1"),
+			s.height-1,
+			s.height,
+		)
+		require.NoError(s.T(), err)
+
+		// Create a mock signaler context for testing.
+		ictx := irrecoverable.NewMockSignalerContext(s.T(), ctx)
+
+		// Start the VersionControl component.
+		s.versionControl.Start(ictx)
+
+		// Ensure the component is ready before proceeding.
+		unittest.RequireComponentsReadyBefore(s.T(), 2*time.Second, s.versionControl)
+
+		scriptExec := NewScriptExecutor(s.log, uint64(0), math.MaxUint64)
+		s.reporter.On("HighestIndexedHeight").Return(s.height+1, nil)
+
+		err = scriptExec.Initialize(s.indexReporter, s.scripts, s.versionControl)
+		s.Require().NoError(err)
+
+		res, err := scriptExec.ExecuteAtBlockHeight(ctx, script, scriptArgs, s.height)
+		s.Assert().Error(err)
```

Comment:
check that we get the specific error that's expected. I believe there is an `ErrorIs()` assertion